<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>938</width>
    <height>474</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Die Türme von Hanoi</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QWidget" name="horizontal" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="spacing">
        <number>6</number>
       </property>
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
       <item>
        <widget class="QFrame" name="horizontalFrame">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <layout class="QHBoxLayout" name="towerLayout">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="TowerWidget" name="tower0" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="focusPolicy">
             <enum>Qt::StrongFocus</enum>
            </property>
            <property name="autoFillBackground">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="TowerWidget" name="tower1" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="focusPolicy">
             <enum>Qt::StrongFocus</enum>
            </property>
            <property name="autoFillBackground">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="TowerWidget" name="tower2" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="focusPolicy">
             <enum>Qt::StrongFocus</enum>
            </property>
            <property name="autoFillBackground">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="label_gameSize">
           <property name="text">
            <string>Anzahl Scheiben:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="gameSize">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="restartGame">
           <property name="text">
            <string>Neues Spiel</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton">
           <property name="text">
            <string>Lösen</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Preferred</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>80</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="label_moveCounter">
           <property name="text">
            <string>Bisher getätigte
Züge:</string>
           </property>
           <property name="textFormat">
            <enum>Qt::PlainText</enum>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="moveCounter">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="mouseTracking">
            <bool>false</bool>
           </property>
           <property name="focusPolicy">
            <enum>Qt::NoFocus</enum>
           </property>
           <property name="acceptDrops">
            <bool>false</bool>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="resetCount">
           <property name="text">
            <string>Zähler
zurücksetzen</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TowerWidget</class>
   <extends>QWidget</extends>
   <header>tower_widget</header>
   <container>1</container>
   <slots>
    <signal>clicked()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>tower0</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>tower0_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>243</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>277</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tower1</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>tower1_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>451</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>483</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tower2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>tower2_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>703</x>
     <y>65</y>
    </hint>
    <hint type="destinationlabel">
     <x>717</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resetCount</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>reset_count()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>918</x>
     <y>299</y>
    </hint>
    <hint type="destinationlabel">
     <x>886</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>restartGame</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>restart_game()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>918</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>934</x>
     <y>121</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>solve_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>825</x>
     <y>102</y>
    </hint>
    <hint type="destinationlabel">
     <x>933</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>tower2_clicked()</slot>
  <slot>tower0_clicked()</slot>
  <slot>tower1_clicked()</slot>
  <slot>reset_count()</slot>
  <slot>restart_game()</slot>
  <slot>solve_clicked()</slot>
 </slots>
</ui>
